src/equations/shallow_water_2d.jl

```diff
@@ -0,0 +1,619 @@
+
+  # Average each factor of products in flux
+  h_v1_avg = 0.5 * (h_v1_ll + h_v1_rr )
+  h_v2_avg = 0.5 * (h_v2_ll + h_v2_rr )
+  v1_avg   = 0.5 * (v1_ll   + v1_rr   )
+  v2_avg   = 0.5 * (v2_ll   + v2_rr   )
+  p_avg    = 0.5 * equations.gravity * h_ll * h_rr
+
+  # Calculate fluxes depending on normal_direction
+  f1 = h_v1_avg * normal_direction[1] + h_v2_avg * normal_direction[2]
+  f2 = f1 * v1_avg + p_avg * normal_direction[1]
+  f3 = f1 * v2_avg + p_avg * normal_direction[2]
+
+  return SVector(f1, f2, f3, zero(eltype(u_ll)))
+end
+
+
+# Calculate maximum wave speed for local Lax-Friedrichs-type dissipation as the
+# maximum velocity magnitude plus the maximum speed of sound
+# TODO: This doesn't really use the `orientation` - should it?
+@inline function max_abs_speed_naive(u_ll, u_rr, orientation::Integer, equations::ShallowWaterEquations2D)
+  h_ll = u_ll[1]
+  _, v1_ll, v2_ll, _ = cons2prim(u_ll, equations)
+  h_rr = u_rr[1]
+  _, v1_rr, v2_rr, _ = cons2prim(u_rr, equations)
+
+  # Calculate velocity magnitude and wave celerity on the left and right
+  v_mag_ll = sqrt(v1_ll^2 + v2_ll^2)
+  c_ll = sqrt(equations.gravity * h_ll)
+  v_mag_rr = sqrt(v1_rr^2 + v2_rr^2)
+  c_rr = sqrt(equations.gravity * h_rr)
+
+  λ_max = max(v_mag_ll, v_mag_rr) + max(c_ll, c_rr)
+end
+
+@inline function max_abs_speed_naive(u_ll, u_rr, normal_direction::AbstractVector, equations::ShallowWaterEquations2D)
+  return max_abs_speed_naive(u_ll, u_rr, 0, equations) * norm(normal_direction)
+end
+
+
+# Calculate minimum and maximum wave speeds for HLL-type fluxes
+@inline function min_max_speed_naive(u_ll, u_rr, orientation::Integer,
+                                     equations::ShallowWaterEquations2D)
+  h_ll = u_ll[1]
+  _, v1_ll, v2_ll, _ = cons2prim(u_ll, equations)
+  h_rr = u_rr[1]
+  _, v1_rr, v2_rr, _ = cons2prim(u_rr, equations)
+
+  if orientation == 1 # x-direction
+    λ_min = v1_ll - sqrt(equations.gravity * h_ll)
+    λ_max = v1_rr + sqrt(equations.gravity * h_rr)
+  else # y-direction
+    λ_min = v2_ll - sqrt(equations.gravity * h_ll)
+    λ_max = v2_rr + sqrt(equations.gravity * h_rr)
+  end
+
+  return λ_min, λ_max
+end
+
+
+@inline function min_max_speed_naive(u_ll, u_rr, normal_direction::AbstractVector,
+                                     equations::ShallowWaterEquations2D)
+  h_ll = u_ll[1]
+  _, v1_ll, v2_ll, _ = cons2prim(u_ll, equations)
+  h_rr = u_rr[1]
+  _, v1_rr, v2_rr, _ = cons2prim(u_rr, equations)
+
+  v_normal_ll = v1_ll * normal_direction[1] + v2_ll * normal_direction[2]
+  v_normal_rr = v1_rr * normal_direction[1] + v2_rr * normal_direction[2]
+
+  norm_ = norm(normal_direction)
+  # The v_normals are already scaled by the norm
+  λ_min = v_normal_ll - sqrt(equations.gravity * h_ll) * norm_
+  λ_max = v_normal_rr + sqrt(equations.gravity * h_rr) * norm_
+
+  return λ_min, λ_max
+end
+
+
+@inline function max_abs_speeds(u, equations::ShallowWaterEquations2D)
+  h = u[1]
+  _, v1, v2, _ = cons2prim(u, equations)
```

Comment:
Just as an idea, since you seem to need just the velocities quite often: Maybe defining a convenience function
```julia
@inline function velocities(u, equations::ShallowWaterEquations2D)
  _, v1, v2, _ = cons2prim(u, equations)
  return v1, v2
end
```
would be nice? That way you could just write
```suggestion
  v1, v2 = velocities(u, equations)
```

Reply:
Why `velocities` instead of `velocity`?

Reply:
Since it's not a scalar value. However, I agree, it's a single quantity, thus velocity velocity is better! In this case, I'd return it as an SVector though

Reply:
Would returning an SVector allocate?

Reply:
No